ci-operator/step-registry/cucushift/upgrade/check/capability/cucushift-upgrade-check-capability-commands.sh

```diff
@@ -0,0 +1,439 @@
+    local expected_status=$2
+    local cvo_field=$3
+
+    cvo_caps=$(oc get clusterversion version -o json | jq -rc "${cvo_field}")
+    if [[ "${capability_set}" == "" ]] && [[ "${cvo_caps}" != "null" ]]; then
+        echo "ERROR: ${expected_status} capability set is empty, but find capabilities ${cvo_caps} in cvo ${cvo_field}"
+        result=1
+    fi
+    if [[ "${capability_set}" != "" ]]; then
+        if [[ "${cvo_caps}" == "null" ]]; then
+            echo "ERROR: ${expected_status} capability set are ${capability_set}, but it's empty in cvo ${cvo_field}"
+            result=1
+        else
+            cvo_caps_str=$(echo $cvo_caps | tr -d '["]' | tr "," " " | xargs -n1 | sort -u | xargs)
+            if [[ "${cvo_caps_str}" == "${capability_set}" ]]; then
+                echo "INFO: ${expected_status} capabilities matches with cvo ${cvo_field}!"
+                echo -e "cvo_caps: ${cvo_caps_str}\n${expected_status} capability set: ${capability_set}"
+            else
+                echo "ERROR: ${expected_status} capabilities does not match with cvo ${cvo_field}!"
+                echo -e "cvo_caps: ${cvo_caps_str}\n${expected_status} capability set: ${capability_set}"
+                echo "diff [cvo_caps] [${expected_status} capability set]"
+                diff <( echo $cvo_caps_str | tr " " "\n" | sort | uniq) <( echo $capability_set | tr " " "\n" | sort | uniq )
+                result=1
+            fi
+        fi
+    fi
+
+    return $result
+} >&2
+
+if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+else
+    echo "Unable to find kubeconfig under ${SHARED_DIR}!"
+    exit 1
+fi
+
+# Setup proxy if it's present in the shared dir
+if [[ -f "${SHARED_DIR}/proxy-conf.sh" ]]; then
+    # shellcheck disable=SC1091
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+echo "RELEASE_IMAGE_LATEST: ${RELEASE_IMAGE_LATEST}"
+echo "RELEASE_IMAGE_TARGET: ${RELEASE_IMAGE_TARGET}"
+
+# echo "RELEASE_IMAGE_LATEST_FROM_BUILD_FARM: ${RELEASE_IMAGE_LATEST_FROM_BUILD_FARM}"
+# export HOME="${HOME:-/tmp/home}"
+# export XDG_RUNTIME_DIR="${HOME}/run"
+# export REGISTRY_AUTH_PREFERENCE=podman # TODO: remove later, used for migrating oc from docker to podman
+# mkdir -p "${XDG_RUNTIME_DIR}"
+# After cluster is set up, ci-operator make KUBECONFIG pointing to the installed cluster,
+# to make "oc registry login" interact with the build farm, set KUBECONFIG to empty,
+# so that the credentials of the build farm registry can be saved in docker client config file.
+# A direct connection is required while communicating with build-farm, instead of through proxy
+# KUBECONFIG="" oc --loglevel=8 registry login
+
+### why use RELEASE_IMAGE_LATEST_FROM_BUILD_FARM???
+#ocp_version=$(oc adm release info ${RELEASE_IMAGE_LATEST_FROM_BUILD_FARM} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
+#echo "OCP Version: $ocp_version"
+
+source_major_version=$( echo "${RELEASE_IMAGE_LATEST}" | cut -f1 -d. )
```

Comment:
If the job is triggered from prow ci cron, the payload is using digest, maybe not tag name, so generally we are using `oc adm release info` to get version.

Reply:
Good catch.

Reply:
Fixed by getting version from cv.status.history.